frontend/src/Battle/index.jsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+import {
+  View, StyleSheet, ScrollView, Text,
+} from 'react-native';
+// import { SERVER_URL } from 'react-native-dotenv';
+import { Toast } from 'native-base';
+import PlanetItem from './PlanetItem';
+import PlayerItem from './PlayerItem';
+import { CardView } from '../common/components';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    paddingTop: -10,
+  },
+  planetsContainer: {
+    marginVertical: 20,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  imageStyle: {
+    height: 64,
+  },
+  scrollView: {
+    flex: 1,
+  },
+});
+
+const PLANETS = ['blue', 'green', 'yellow', 'red', 'purple'];
+
+const mockedUsers = [
+  {
+    _id: '5e12d061f22b9d30a8ac7507',
+    username: 'testuser3',
+    hp: 0,
+    attack: 0,
+    defence: 0,
+    allLevels: 0,
+    planetList: [
+      'green',
+      'blue',
+      'yellow',
+    ],
+  },
+  {
+    _id: '5e18181bc5cbf11a8870837e',
+    username: 'clay',
+    hp: 64,
+    attack: 177,
+    defence: 177,
+    allLevels: 276,
+    planetList: [
+      'yellow',
+    ],
+  },
+  {
+    _id: '5e1b1689f09b912f0461ed61',
+    username: 'wind',
+    hp: 0,
+    attack: 0,
+    defence: 0,
+    allLevels: 0,
+    planetList: [
+      'yellow',
+    ],
+  },
+  {
+    _id: '5e1b23104320f53d4c8593f3',
+    username: 'jialu',
+    hp: 0,
+    attack: 0,
+    defence: 0,
+    allLevels: 0,
+    planetList: [
+      'yellow',
+    ],
+  },
+];
+
+function Battle() {
+  const [active, setActive] = useState(null);
+  const [players] = useState(mockedUsers);
+
+  function handleBattleStart(id, name) {
+    Toast.show({
+      text: `You are going to battle with ${name.toUpperCase()}`,
+    });
+  }
+
+  // useEffect(() => {
```

Comment:
this is for connecting the backend right?

Reply:
Yes, just waiting for the backend to be merged and I can start the connect work quickly.

Reply:
battle map backend is merged